File: javascript/array/500. Keyboard Row.js
/**
 * @param {string[]} words
 * @return {string[]}
 */
var findWords = function(words) {
    const rows = ["qwertyuiop", "asdfghjkl", "zxcvbnm"];
    const res = [];
    let cnt = "";

    words.forEach(word => {
        for (let c of word) {
            rows.forEach((row, index) => {
                if (row.includes(c.toLowerCase())) { cnt += `${index}`; }
            }) 
        }
        if (new Set([...cnt]).size === 1) { res.push(word) };
        cnt = "";
    })
    return res;
};

File: python/Array_Hash/500. Keyboard Row.py
'''
own
Time: O(w * c)  -- w: # of words; c: length of word
Space: O(w)
'''

class Solution:
    def findWords(self, words: List[str]) -> List[str]:
        row1, row2, row3 = "qwertyuiop", "asdfghjkl", "zxcvbnm"
        cnt = ""
        res = []

        for w in words:
            for c in w:
                if c.lower() in row1:
                    cnt += "1"
                elif c.lower() in row2:
                    cnt += "2"
                else:
                    cnt +="3" 
            if len(set(cnt)) == 1:
                res.append(w)
            cnt = ""

        return res